### PR TITLE
Refactor navigation endpoint

### PR DESCRIPTION
**Pull request by sneridagh**
Includes the new nav_title attribute

### CODE REVIEW

File: src/plone/restapi/tests/http-examples/translated_messages_types_folder.resp [hunk 1]
```diff
@@ -252,8 +252,7 @@ Content-Type: application/json+schema
     }
   }, 
   "required": [
-    "title", 
-    "nextPreviousEnabled"
```
Comment (sneridagh) [inline src/plone/restapi/tests/http-examples/translated_messages_types_folder.resp]: I have no clue how this got removed

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
@tisto is green now... although I saw Jenkins being very flaky

**Comment by tisto** [comment]
@jensens would you like to have a look at this refactoring of the nav endpoint (since you recommended it in the first place) :)

**Review by tisto — APPROVED** [review]

**Review by tiberiuichim — APPROVED** [review]
>            self.customize_entry(entry, brain)

nice to have that

**Pull request merged**